var/spack/repos/builtin/packages/py-psyclone/package.py

```diff
@@ -0,0 +1,65 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+from spack import *
+
+
+class PyPsyclone(PythonPackage):
+    """Code generation for the PSyKAl framework from the GungHo project,
+       as used by the LFRic model at the UK Met Office."""
+
+    homepage = "https://github.com/stfc/PSyclone"
+    url      = "https://github.com/stfc/PSyclone/archive/1.5.1.tar.gz"
+    giturl   = "https://github.com/stfc/PSyclone.git"
+
+    version('1.5.1', git=giturl,
+            commit='eba7a097175b02f75dec70616cf267b7b3170d78')
+    version('develop', git=giturl, branch='master')
+
+    depends_on('py-setuptools', type='build')
+
+    depends_on('python', type=('build', 'run'))
+    depends_on('py-pyparsing', type=('build', 'run'))
+
+    # Test cases fail without compatible versions of py-fparser:
+    depends_on('py-fparser@0.0.5', type=('build', 'run'), when='@1.5.1')
+    depends_on('py-fparser', type=('build', 'run'), when='@1.5.2:')
+
+    # Use type='test' when available:
```

Comment:
The test deptype is now available.

Reply:
I switched to `type='test'` for selected dependencies, and it worked fine (unlike py-fparser at commit 3a873df).